File: VB/CS/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB

Namespace SelfReferenced

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            XpoDefault.DataLayer = XpoDefault.GetDataLayer(AutoCreateOption.DatabaseAndSchema)
            CreateDefaultData()
            InitializeComponent()
        End Sub

        Private Sub CreateDefaultData()
            Using uow As UnitOfWork = New UnitOfWork()
                If uow.FindObject(Of Employee)(Nothing) IsNot Nothing Then Return
                Dim AndrewFuller As Employee = New Employee(uow)
                AndrewFuller.FullName = "Andrew Fuller"
                AndrewFuller.Title = "Vice President, Sales"
                Dim NancyDavolio As Employee = New Employee(uow)
                NancyDavolio.FullName = "Nancy Davolio"
                NancyDavolio.Title = "Sales Representative"
                NancyDavolio.Manager = AndrewFuller
                Dim JanetLeverling As Employee = New Employee(uow)
                JanetLeverling.FullName = "Janet Leverling"
                JanetLeverling.Title = "Sales Representative"
                JanetLeverling.Manager = AndrewFuller
                Dim MargaretPeacock As Employee = New Employee(uow)
                MargaretPeacock.FullName = "Margaret Peacock"
                MargaretPeacock.Title = "Sales Representative"
                MargaretPeacock.Manager = AndrewFuller
                Dim StevenBuchanan As Employee = New Employee(uow)
                StevenBuchanan.FullName = "Steven Buchanan"
                StevenBuchanan.Title = "Sales Manager"
                StevenBuchanan.Manager = AndrewFuller
                Dim MichaelSuyama As Employee = New Employee(uow)
                MichaelSuyama.FullName = "Michael Suyama"
                MichaelSuyama.Title = "Sales Representative"
                MichaelSuyama.Manager = StevenBuchanan
                Dim RobertKing As Employee = New Employee(uow)
                RobertKing.FullName = "Robert King"
                RobertKing.Title = "Sales Representative"
                RobertKing.Manager = StevenBuchanan
                Dim LauraCallahan As Employee = New Employee(uow)
                LauraCallahan.FullName = "Laura Callahan"
                LauraCallahan.Title = "Inside Sales Coordinator"
                LauraCallahan.Manager = AndrewFuller
                Dim AnneDodsworth As Employee = New Employee(uow)
                AnneDodsworth.FullName = "Anne Dodsworth"
                AnneDodsworth.Title = "Sales Representative"
                AnneDodsworth.Manager = StevenBuchanan
                uow.CommitChanges()
            End Using
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            treeList1.KeyFieldName = "This"
            treeList1.ParentFieldName = "Manager!"
            treeList1.RootValue = Nothing
            treeList1.DataSource = xpCollection1
            treeList1.ExpandAll()
        End Sub

        Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            If unitOfWork1.InTransaction Then unitOfWork1.CommitChanges()
        End Sub
    End Class
End Namespace

File: CS/CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Data.Filtering;

namespace SelfReferenced {
    public partial class Form1 : Form {
        public Form1() {
            XpoDefault.DataLayer = XpoDefault.GetDataLayer(AutoCreateOption.DatabaseAndSchema);
            CreateDefaultData();

            InitializeComponent();
        }

        void CreateDefaultData() {
            using(UnitOfWork uow = new UnitOfWork()) {
                if(uow.FindObject<Employee>(null) != null) return;

                Employee AndrewFuller = new Employee(uow);
                AndrewFuller.FullName = "Andrew Fuller";
                AndrewFuller.Title = "Vice President, Sales";

                Employee NancyDavolio = new Employee(uow);
                NancyDavolio.FullName = "Nancy Davolio";
                NancyDavolio.Title = "Sales Representative";
                NancyDavolio.Manager = AndrewFuller;

                Employee JanetLeverling = new Employee(uow);
                JanetLeverling.FullName = "Janet Leverling";
                JanetLeverling.Title = "Sales Representative";
                JanetLeverling.Manager = AndrewFuller;

                Employee MargaretPeacock = new Employee(uow);
                MargaretPeacock.FullName = "Margaret Peacock";
                MargaretPeacock.Title = "Sales Representative";
                MargaretPeacock.Manager = AndrewFuller;

                Employee StevenBuchanan = new Employee(uow);
                StevenBuchanan.FullName = "Steven Buchanan";
                StevenBuchanan.Title = "Sales Manager";
                StevenBuchanan.Manager = AndrewFuller;

                Employee MichaelSuyama = new Employee(uow);
                MichaelSuyama.FullName = "Michael Suyama";
                MichaelSuyama.Title = "Sales Representative";
                MichaelSuyama.Manager = StevenBuchanan;

                Employee RobertKing = new Employee(uow);
                RobertKing.FullName = "Robert King";
                RobertKing.Title = "Sales Representative";
                RobertKing.Manager = StevenBuchanan;

                Employee LauraCallahan = new Employee(uow);
                LauraCallahan.FullName = "Laura Callahan";
                LauraCallahan.Title = "Inside Sales Coordinator";
                LauraCallahan.Manager = AndrewFuller;

                Employee AnneDodsworth = new Employee(uow);
                AnneDodsworth.FullName = "Anne Dodsworth";
                AnneDodsworth.Title = "Sales Representative";
                AnneDodsworth.Manager = StevenBuchanan;

                uow.CommitChanges();
            }
        }

        private void Form1_Load(object sender, EventArgs e) {
            treeList1.KeyFieldName = "This";
            treeList1.ParentFieldName = "Manager!";
            treeList1.RootValue = null;
            treeList1.DataSource = xpCollection1;

            treeList1.ExpandAll();
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e) {
            if(unitOfWork1.InTransaction)
                unitOfWork1.CommitChanges();
        }
    }
}